operator/src/main/java/org/keycloak/operator/controllers/KeycloakDeploymentConfig.java

```diff
@@ -165,7 +153,11 @@ public void configureTLS() {
         var userRelativePath = readConfigurationValue(Constants.KEYCLOAK_HTTP_RELATIVE_PATH_KEY);
         var kcRelativePath = (userRelativePath == null) ? "" : userRelativePath;
         var protocol = (keycloakCR.getSpec().isHttp()) ? "http" : "https";
-        var kcPort = (keycloakCR.getSpec().isHttp()) ? Constants.KEYCLOAK_HTTP_PORT : Constants.KEYCLOAK_HTTPS_PORT;
+
+        var kcPort = keycloakCR.getSpec().getHostnameSpec().getPort();
+        if (kcPort == null) {
+            kcPort = keycloakCR.getSpec().isHttp() ? Constants.KEYCLOAK_HTTP_PORT : Constants.KEYCLOAK_HTTPS_PORT;
```

Comment:
@vmuzikar In the description of the associated issue is mentioned that the `INSECURE-DISABLE` functionality should be removed. Does it mean that we're gonna support only the production-ready environment with Operator? Will there be no support from the developer's perspective?

Reply:
jFTR – this was discussed offline and will be reflected in docs.